### PR TITLE
Prevent misleading error message on initial VNC connection failures

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

**Comment by okurz** [comment]
@asmorodskyi @foursixnine

File: consoles/VNC.pm [hunk 1]
```diff
@@ -131,14 +131,23 @@ sub login {
...
             PeerPort => $port,
             Proto    => 'tcp',
         );
         if (!$socket) {
+            $err_cnt++;
             return if (time > $endtime);
+            # we might be too fast trying to connect to the VNC host (e.g.
+            # qemu) so ignore the first three occurences of a failed
+            # connection attempt.
+            if ($err_cnt < 3) {
```
Comment (asmorodskyi) [inline consoles/VNC.pm]: you should just  write 
if ($err_cnt < 3) {
bmwqemu::diag "Error connecting to host <$hostname>: $@";
}

and remove duplicated lines 
sleep 1;
next;

**Comment by asmorodskyi** [comment]
LGTM

File: consoles/VNC.pm [hunk 2]
```diff
@@ -131,15 +131,20 @@ sub login {
...
             Proto    => 'tcp',
         );
         if (!$socket) {
+            $err_cnt++;
             return if (time > $endtime);
-            bmwqemu::diag "Error connecting to host <$hostname>: $@";
+            # we might be too fast trying to connect to the VNC host (e.g.
+            # qemu) so ignore the first occurences of a failed
+            # connection attempt.
+            bmwqemu::diag "Error connecting to host <$hostname>: $@" if ($err_cnt > 2);
```
Comment (coolo) [inline consoles/VNC.pm]: for next time: you don't need the extra () if the condition is simple and post (it shouldn't be post if it's not simple - anyway).

**Comment by coolo** [comment]
how do you defend the magic number 2?

**Comment by okurz** [comment]
On March 11, 2017 10:58:47 AM GMT+01:00, Stephan Kulow <notifications@github.com> wrote:
>how do you defend the magic number 2?

Yeah, hard to defend. Changed to variable with descriptive name and comment

**Pull request merged**